### PR TITLE
Memory Optimizations

### PR DESCRIPTION
**Pull request by ayushaga14**

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -463,16 +471,24 @@ func run(handle *pcap.Handle, apiCollectionId int, source string) {
...
+			if bytesIn > bytesInThreshold {
+				log.Println("exceeded bytesInThreshold: ", bytesInThreshold, " with curr: ", bytesIn)
+				log.Println("limit reached")
+				wipeOut()
+				bytesIn = 0
+				bytesInEpoch = time.Now()
+				log.Println("sleep time: ", time.Now())
+				time.Sleep(20 * time.Second)
+				kafkaWriter.Close()
+				kafkaWriter = gomiddleware.GetKafkaWriter(kafka_url, "akto.api.logs", kafka_batch_size, kafka_batch_time_secs_duration*time.Second)
```
Comment (avneesh-akto) [inline main.go]: why is it being initialised again?

**Review by avneesh-akto — APPROVED** [review]

File: main.go [hunk 2]
```diff
@@ -247,6 +247,8 @@ func tryReadFromBD(bd *bidi, isPending bool) {
 		return
 	}
 
+	log.Println("len(requests): ", len(requests))
```
Comment (ankush-jain-akto) [inline main.go]: Remove log

File: main.go [hunk 3]
```diff
@@ -38,7 +38,7 @@ import (
 )
 
 var printCounter = 500
-var bytesInThreshold = 10 * 1024 * 1024
+var bytesInThreshold = 20 * 1024 * 1024
```
Comment (ankush-jain-akto) [inline main.go]: Keep this same.

File: main.go [hunk 4]
```diff
@@ -463,16 +466,22 @@ func run(handle *pcap.Handle, apiCollectionId int, source string) {
 
 			bytesIn += len(tcp.Payload)
 
+			if bytesIn > bytesInThreshold {
+				log.Println("exceeded bytesInThreshold: ", bytesInThreshold, " with curr: ", bytesIn)
+				log.Println("limit reached, sleeping", time.Now())
+				wipeOut()
+				bytesIn = 0
+				bytesInEpoch = time.Now()
+				time.Sleep(20 * time.Second)
```
Comment (ankush-jain-akto) [inline main.go]: Change this 10 seconds?

**Pull request merged**